### PR TITLE
Pass thread state slice's callstack down to the client

### PR DESCRIPTION
**Pull request by florian-kuebler**
For CaptureClientEvents with thread state slices that have a callstack id associated with it, we are now storing the callstack id in the ThreadStateSliceInfo.

We are also using the same de-duplication of the callstacks on the client that we are using for callstack events.

Test: Unit test
Bug: http://b/235554760

### CODE REVIEW

File: src/ClientData/include/ClientData/ThreadStateSliceInfo.h [hunk 1]
```diff
@@ -62,6 +69,7 @@ class ThreadStateSliceInfo {
   WakeupReason wakeup_reason_;
   uint32_t wakeup_tid_;
   uint32_t wakeup_pid_;
+  std::optional<uint64_t> triggering_callstack_id_;
```
Comment (dpallotti) [inline src/ClientData/include/ClientData/ThreadStateSliceInfo.h:72]: I would prefer the name `switch_out_or_wakeup_callstack_id` that we use in the protos.

Comment (dpallotti) [reply src/ClientData/include/ClientData/ThreadStateSliceInfo.h:72]: Same in all the other places in this change.

File: src/CaptureClient/CaptureEventProcessorTest.cpp [hunk 2]
```diff
@@ -1170,6 +1181,110 @@ TEST(CaptureEventProcessor, CanHandleThreadStateSlices) {
...
+
+  ClientCaptureEvent interned_callstack_event;
+  InternedCallstack* interned_callstack = interned_callstack_event.mutable_interned_callstack();
+  interned_callstack->set_key(kCallstackId);
+  interned_callstack->mutable_intern()->add_pcs(kFrame1);
+  interned_callstack->mutable_intern()->add_pcs(kFrame2);
+  interned_callstack->mutable_intern()->add_pcs(kFrame3);
+  interned_callstack->mutable_intern()->set_type(orbit_grpc_protos::Callstack::kComplete);
+
+  ClientCaptureEvent runnable_event;
```
Comment (dpallotti) [inline src/CaptureClient/CaptureEventProcessorTest.cpp:1245]: Name is a bit strange: `runnable_thread_state_slice_event` is more fitting with the naming pattern.

**Review by dpallotti — APPROVED** [review]

File: src/ClientData/include/ClientData/ThreadStateSliceInfo.h (hunk 1 above)
Comment (florian-kuebler) [reply src/ClientData/include/ClientData/ThreadStateSliceInfo.h:72]: Done.

**Pull request merged**